rust/agents/relayer/src/msg/pending_message.rs

```diff
@@ -329,21 +363,41 @@ impl PendingMessage {
     }
 
     fn reset_attempts(&mut self) {
-        self.num_retries = 0;
+        self.set_retries(0);
         self.next_attempt_after = None;
         self.last_attempted_at = Instant::now();
     }
 
     fn inc_attempts(&mut self) {
-        self.num_retries += 1;
+        self.set_retries(self.num_retries + 1);
         self.last_attempted_at = Instant::now();
         self.next_attempt_after = PendingMessage::calculate_msg_backoff(self.num_retries)
             .map(|dur| self.last_attempted_at + dur);
     }
 
+    fn set_retries(&mut self, retries: u32) {
+        self.num_retries = retries;
+        self.persist_retries();
+    }
+
+    fn persist_retries(&self) {
+        if let Err(e) = self
+            .ctx
+            .origin_db
+            .store_pending_message_data_by_message_id(&self.message.id(), &self.num_retries)
+        {
+            trace!(
+                "No persisted retry count for message {}. Hyperlane DB error: {}",
```

Comment:
Think this would be fair as `info` or `warn`, and should say that the persisting of the retry count failed instead of saying that there was no persisted retry count